db/data/20220331152016_fix_lead_schools_not_applicable.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+class FixLeadSchoolsNotApplicable < ActiveRecord::Migration[6.1]
+  def up
+    def fix_trainee(trainee)
+      lead_school_dttp_id = trainee.dttp_trainee&.latest_placement_assignment&.lead_school_id
+      lead_school_urn = Dttp::School.find_by(dttp_id: lead_school_dttp_id)&.urn
+
+      trainee.update(lead_school_not_applicable: true) if %w[900010 99999996].include?(lead_school_urn)
+
+      # It seems if we remove the 'zzz', the URN is actually valid and school records exist
+      trainee.update(lead_school: School.find_by(urn: lead_school_urn.remove("zzz"))) if lead_school_urn&.include?("zzz")
+
+      # The following URN's don't have a matching school but when checked against
+      # https://get-information-schools.service.gov.uk/Search we find school names that do exist in the schools table
+      # with a different URN.
+      trainee.update(lead_school: School.find_by(name: "Chellaston Academy")) if lead_school_urn == "148639"
+      trainee.update(lead_school: School.find_by(name: "Greentrees Primary School")) if lead_school_urn == "148680"
+      trainee.update(lead_school: School.find_by(name: "Redwell Primary School")) if lead_school_urn == "148611"
+    end
+
+    dttp_trainees = Trainee.created_from_dttp.where(lead_school_id: nil)
+
+    # Teaching apprenticeships that are missing a lead school
+    dttp_trainees.pg_teaching_apprenticeship.where.not(employing_school_id: nil).each { |trainee| fix_trainee(trainee) }
+
+    # School Direct (fee funded) that might have a lead school that is not applicable but won't have an employing school
+    dttp_trainees.school_direct_tuition_fee.where(employing_school_id: nil).each { |trainee| fix_trainee(trainee) }
```

Comment:
I think we still need to do school_direct_salaried as they have a lead school too, or at least should have.

We need set lead_school on salaried or tuition_fee where lead school is nil. I expect employing_school might have the same problem but I don't think we should only be fixing where employing_school is nil, it should be where lead_school is nil.

Reply:
I've pushed another update. Local testing shows that there was a number school direct trainees that didn't have a `lead_school_id` nor `lead_school_not_applicable` set to true. After running the migration, this was fixed.